test/ReactSetStateTestComponent2.js

```diff
@@ -71,8 +71,15 @@ class ReactSetStateTestComponent2 extends React.Component {
         this.recordStateChange(this.state.counter);
     }
 
+    static getDerivedStateFromProps(nextProps, __) {
+        _shouldUpdate = nextProps.shouldUpdate;
+        _shouldUpdate ? _component2UpdatingSetStateLifeCycleCalls.push("getDerivedStateFromProps") : _component2NonUpdatingSetStateLifeCycleCalls.push("getDerivedStateFromProps");
```

Comment:
#nit this line is pretty long, could get wrapped
```suggestion
        _shouldUpdate
            ? _component2UpdatingSetStateLifeCycleCalls.push("getDerivedStateFromProps") 
            : _component2NonUpdatingSetStateLifeCycleCalls.push("getDerivedStateFromProps");
```